internal/provider/resource_cluster.go

```diff
@@ -877,6 +938,20 @@ func (r *ClusterResource) Update(ctx context.Context, req resource.UpdateRequest
 		tflog.Info(ctx, "cluster resources updated")
 	}
 
+	// Get the latest cluster state and save it to state
+	now, err := r.client.GetClusterByNsID(ctx, nsID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read cluster",
+			err.Error(),
+		)
+		return
+	}
+	resp.Diagnostics.Append(clusterToDataModel(now, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
```

Comment:
I don't see BYOC `ClusterName` covered in the Update API, is this expected?

`ClusterName` is immutable and we should return error if a diff is found.